src/main/java/com/maxmind/geoip2/model/AbstractCityResponse.java

```diff
@@ -36,7 +37,27 @@
         this.subdivisions = subdivisions != null ? subdivisions : new ArrayList<>();
     }
 
-    protected static ArrayList<Subdivision> mapSubdivisions(
+    AbstractCityResponse(
+            AbstractCityResponse response,
+            String ipAddress,
+            Network network,
+            List<String> locales
+    ) {
+        this(
+            response.getCity() != null ? new City(response.getCity(), locales) : null,
```

Comment:
Under what case can this and the ones below be `null`? I thought we would set it to the empty object rather than making it `null`.

Reply:
If `city` wasn't found in the data, it'll be `null` after decoding I believe. Since we call the other constructor here with `null`, that other constructor creates the empty object

Reply:
The latest commit makes this clearer now I think.